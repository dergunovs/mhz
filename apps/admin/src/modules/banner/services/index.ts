import { ComputedRef, Ref } from 'vue';

import { api, useQuery, useMutation, IPageQuery, convertParams } from 'mhz-helpers';
import { API_BANNER, IBanner, IBaseReply, IFilterData } from 'mhz-contracts';

export function getBanners(query: Ref<IPageQuery | number>) {
  async function fn() {
    const params = convertParams(query);

    const { data } = await api.get<{ data: IBanner[]; total: number; filters: IFilterData }>(API_BANNER, { params });

    return data;
  }

  return useQuery({ queryKey: [API_BANNER, query], queryFn: fn });
}

export function getBanner(id?: ComputedRef<string | string[]>) {
  async function fn() {
    if (!id?.value) return null;

    const { data } = await api.get<{ data: IBanner }>(`${API_BANNER}/${id.value}`);

    return data.data;
  }

  return useQuery({ queryKey: [API_BANNER, id], queryFn: fn });
}

export function postBanner(options: object) {
  async function fn(formData: IBanner) {
    const { data } = await api.post<IBaseReply>(API_BANNER, formData);

    return data;
  }

  return useMutation({ mutationKey: [API_BANNER], mutationFn: fn, ...options });
}

export function updateBanner(options: object) {
  async function fn(formData: IBanner) {
    const { data } = await api.patch<IBaseReply>(`${API_BANNER}/${formData._id}`, formData);

    return data;
  }

  return useMutation({ mutationKey: [API_BANNER], mutationFn: fn, ...options });
}

export function deleteBanner(options: object) {
  async function fn(id?: string) {
    if (!id) return null;

    const { data } = await api.delete<IBaseReply>(`${API_BANNER}/${id}`);

    return data;
  }

  return useMutation({ mutationKey: [API_BANNER], mutationFn: fn, ...options });
}
