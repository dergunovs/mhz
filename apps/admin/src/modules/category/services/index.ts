import { ComputedRef } from 'vue';

import { api, useQuery, useMutation } from 'mhz-helpers';
import { API_CATEGORY, ICategory, IBaseReply } from 'mhz-contracts';

export function getCategories(options?: object) {
  async function fn() {
    const { data } = await api.get<{ data: ICategory[] }>(API_CATEGORY);

    return data.data;
  }

  return useQuery({ queryKey: [API_CATEGORY], queryFn: fn, ...options });
}

export function getCategory(id?: ComputedRef<string | string[]>) {
  async function fn() {
    if (!id?.value) return null;

    const { data } = await api.get<{ data: ICategory }>(`${API_CATEGORY}/${id.value}`);

    return data.data;
  }

  return useQuery({ queryKey: [API_CATEGORY, id], queryFn: fn });
}

export function postCategory(options: object) {
  async function fn(formData: ICategory) {
    const { data } = await api.post<IBaseReply>(API_CATEGORY, formData);

    return data;
  }

  return useMutation({ mutationKey: [API_CATEGORY], mutationFn: fn, ...options });
}

export function updateCategory(options: object) {
  async function fn(formData: ICategory) {
    const { data } = await api.patch<IBaseReply>(`${API_CATEGORY}/${formData._id}`, formData);

    return data;
  }

  return useMutation({ mutationKey: [API_CATEGORY], mutationFn: fn, ...options });
}

export function deleteCategory(options: object) {
  async function fn(id?: string) {
    if (!id) return null;

    const { data } = await api.delete<IBaseReply>(`${API_CATEGORY}/${id}`);

    return data;
  }

  return useMutation({ mutationKey: [API_CATEGORY], mutationFn: fn, ...options });
}
