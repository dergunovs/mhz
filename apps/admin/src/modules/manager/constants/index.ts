export const URL_MANAGER = '/managers';
