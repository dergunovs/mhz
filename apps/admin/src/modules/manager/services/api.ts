import { ComputedRef, Ref } from 'vue';

import { api, IPageQuery, convertParams } from 'mhz-helpers';
import { API_MANAGER, IBaseReply, IManager } from 'mhz-contracts';

export async function getManagersApi(query: Ref<IPageQuery | number>) {
  const params = convertParams(query);

  const { data } = await api.get<{ data: IManager[]; total: number }>(API_MANAGER, { params });

  return data;
}

export async function getManagerApi(id?: ComputedRef<string | string[]>) {
  if (!id?.value) return null;

  const { data } = await api.get<{ data: IManager }>(`${API_MANAGER}/${id.value}`);

  return data.data;
}

export async function postManagerApi(formData: IManager) {
  const { data } = await api.post<IBaseReply>(API_MANAGER, formData);

  return data;
}

export async function updateManagerApi(formData: IManager) {
  const { data } = await api.patch<IBaseReply>(`${API_MANAGER}/${formData._id}`, formData);

  return data;
}

export async function deleteManagerApi(id?: string) {
  if (!id) return null;

  const { data } = await api.delete<IBaseReply>(`${API_MANAGER}/${id}`);

  return data;
}
