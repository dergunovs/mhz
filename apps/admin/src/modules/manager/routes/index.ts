import { URL_MANAGER, URL_MANAGER_CREATE, URL_MANAGER_EDIT } from '@/manager/constants';

export const managerRoutes = [
  {
    path: URL_MANAGER,
    name: 'ManagerList',
    component: () => import('@/manager/pages/ManagerListPage.vue'),
  },
  {
    path: URL_MANAGER_CREATE,
    name: 'ManagerCreate',
    component: () => import('@/manager/pages/ManagerCreatePage.vue'),
  },
  {
    path: `${URL_MANAGER_EDIT}/:manager`,
    name: 'ManagerEdit',
    component: () => import('@/manager/pages/ManagerEditPage.vue'),
  },
];
