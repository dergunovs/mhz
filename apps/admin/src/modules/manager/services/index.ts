import { ComputedRef, Ref } from 'vue';

import { IManager } from 'mhz-types';
import { api, useMutation, useQuery, IPageQuery } from 'mhz-helpers';

import { API_MANAGER } from '@/manager/constants';

export function getManagers(query: Ref<IPageQuery | number>) {
  async function fn(): Promise<{ data: IManager[]; total: number }> {
    const params =
      typeof query.value === 'number'
        ? { page: query.value }
        : {
            page: query.value.page || 1,
            sort: query.value.sort.value,
            dir: query.value.sort.isAsc === false ? 'desc' : 'asc',
          };

    const { data } = await api.get(API_MANAGER, { params });

    return data;
  }

  return useQuery({ queryKey: [API_MANAGER, query], queryFn: fn });
}

export function getManager(id?: ComputedRef<string | string[]>) {
  async function fn(): Promise<IManager | null> {
    if (!id?.value) return null;

    const { data } = await api.get(`${API_MANAGER}/${id.value}`);

    return data;
  }

  return useQuery({ queryKey: [API_MANAGER, id], queryFn: fn, refetchOnMount: true });
}

export function postManager(options: object) {
  async function fn(formData: IManager) {
    await api.post(API_MANAGER, formData);
  }

  return useMutation({ mutationKey: [API_MANAGER], mutationFn: fn, ...options });
}

export function updateManager(id: ComputedRef<string | undefined>, options: object) {
  async function fn(formData: IManager) {
    await api.patch(`${API_MANAGER}/${id.value}`, formData);
  }

  return useMutation({ mutationKey: [API_MANAGER, id], mutationFn: fn, ...options });
}

export function deleteManager(options: object) {
  async function fn(id?: string) {
    await api.delete(`${API_MANAGER}/${id}`);
  }

  return useMutation({ mutationKey: [API_MANAGER], mutationFn: fn, ...options });
}
