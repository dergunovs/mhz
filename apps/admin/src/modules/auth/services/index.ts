import { Ref } from 'vue';
import { useMutation, useQuery } from '@tanstack/vue-query';

import { ILoginFormData } from '@/auth/interface';
import { API_GET_CHECK_AUTH, API_POST_LOGIN } from '@/auth/constants';

import { api } from '@/common/services/api';

export function postLogin(formData: Ref<ILoginFormData>, options: object) {
  async function fn(): Promise<string | undefined> {
    const { data } = await api.post(API_POST_LOGIN, formData.value);

    return data.token;
  }

  return useMutation({
    mutationKey: [API_POST_LOGIN, formData],
    mutationFn: fn,
    ...options,
  });
}

export function getCheckAuth(options: object) {
  async function fn() {
    const { data } = await api.get(API_GET_CHECK_AUTH);

    return data;
  }

  return useQuery({
    queryKey: [API_GET_CHECK_AUTH],
    queryFn: fn,
    ...options,
  });
}
