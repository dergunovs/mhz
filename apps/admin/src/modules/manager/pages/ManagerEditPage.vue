<template>
  <div>
    <PageTitle :links="links">{{ title }}</PageTitle>

    <div v-if="manager?.dateLoggedIn" :class="$style.login">
      <b>Login date:</b> {{ formatDateTime(manager?.dateLoggedIn) }}
    </div>

    <ManagerForm v-if="manager" :manager="manager" data-test="manager-edit-page-form" />
  </div>
</template>

<script setup lang="ts">
import { computed } from 'vue';
import { useRoute } from 'vue-router';
import { useHead } from '@unhead/vue';

import { formatDateTime } from 'mhz-helpers';

import PageTitle from '@/layout/components/PageTitle.vue';
import ManagerForm from '@/manager/components/ManagerForm.vue';

import { getManager } from '@/manager/services';
import { URL_MANAGER } from '@/manager/constants';
import { URL_MAIN } from '@/common/constants';

const route = useRoute();

const managerId = computed(() => route.params.manager);

const { data: manager } = getManager(managerId);

const title = 'Edit manager';

const links = computed(() => [
  { url: URL_MAIN, title: 'Main' },
  { url: URL_MANAGER, title: 'Managers' },
  { url: route.path, title: manager.value ? `${manager.value.firstName} ${manager.value.lastName}` : '' },
]);

useHead({
  title,
});
</script>

<style module lang="scss">
.login {
  text-align: right;
}
</style>
