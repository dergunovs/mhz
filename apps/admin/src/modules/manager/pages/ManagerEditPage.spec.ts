import { DefineComponent } from 'vue';
import { describe, it, expect, beforeEach, afterEach, vi } from 'vitest';
import { VueWrapper, enableAutoUnmount } from '@vue/test-utils';
import { IManager } from 'mhz-contracts';
import { dataTest } from 'mhz-helpers';

import ManagerEditPage from './ManagerEditPage.vue';

import { mockQueryReply, wrapperFactory } from '@/common/test';
import * as managerServices from '@/manager/services';
import { MANAGERS } from '@/manager/fixtures';

const MANAGER = MANAGERS.data[0];

const spyGetManager = vi.spyOn(managerServices, 'getManager').mockReturnValue(mockQueryReply(MANAGER));

const managerEditPageForm = dataTest('manager-edit-page-form');

let wrapper: VueWrapper;

beforeEach(() => {
  wrapper = wrapperFactory(ManagerEditPage, {});
});

enableAutoUnmount(afterEach);

describe('ManagerEditPage', async () => {
  it('exists', async () => {
    expect(wrapper.findComponent(ManagerEditPage)).toBeTruthy();
  });

  it('gets manager and passes it to form props', async () => {
    expect(spyGetManager).toBeCalledTimes(1);

    expect(
      wrapper.findComponent<DefineComponent<{ manager: IManager }>>(managerEditPageForm).vm.$props.manager
    ).toEqual(MANAGER);
  });
});
