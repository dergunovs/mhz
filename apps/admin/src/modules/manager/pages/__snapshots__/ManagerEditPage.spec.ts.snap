// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ManagerEditPage > matches snapshot 1`] = `
"<div>
  <page-title-stub links="[object Object],[object Object],[object Object]"></page-title-stub>
  <div class="_login_787d7f"><b>Login date:</b> Jan 18, 2024, 2:32 PM</div>
  <manager-form-stub manager="[object Object]" data-test="manager-edit-page-form"></manager-form-stub>
</div>"
`;
